asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py

```diff
@@ -0,0 +1,875 @@
+                    args.mols,
+                    "sdf",
+                    return_mols=True,
+                    smiles_as_title=args.smiles_as_title,
+                )
+                used_3d = True
+                logger.info("setting used_3d to True")
+            else:
+                logger.info("Using 2D representation from SDF file")
+                exp_data = oe_load_exp_from_file(
+                    args.mols, "sdf", smiles_as_title=args.smiles_as_title
+                )
+
+        else:
+            raise ValueError(
+                f"Input molecules must be a SMILES file, SDF file, or SMILES string. Got {args.mols}"
+            )
+
+    logger.info(f"Loaded {len(exp_data)} molecules.")
+    if len(exp_data) == 0:
+        logger.error("No molecules loaded.")
+        raise ValueError("No molecules loaded.")
+
+    if args.verbose:
+        # we could make this just a debug statement but avoid looping over all molecules if not needed
+        for exp in exp_data:
+            logger.debug(f"Loaded molecule {exp.compound_id}: {exp.smiles}")
+
+    #########################
+    #     protein prep      #
+    #########################
+
+    # parse prep arguments
+    prep_dir = output_dir / "prep"
+    prep_dir.mkdir(parents=True, exist_ok=True)
+    intermediate_files.append(prep_dir)
+
+    logger.info(f"Prepping receptor in {prep_dir} at {datetime.now().isoformat()}")
+
+    # check inputs
+    receptor = Path(args.receptor)
+    if receptor.suffix != ".pdb":
+        raise ValueError(f"Receptor must be a PDB file. Got {args.receptor}")
+
+    if not receptor.exists():
+        raise ValueError(f"Receptor file does not exist: {args.receptor}")
+
+    if args.ref_prot is not None:
+        ref_prot = Path(args.ref_prot)
+        if ref_prot.suffix != ".pdb":
+            raise ValueError(
+                f"Reference protein must be a PDB file. Got {args.ref_prot}"
+            )
+
+    if args.loop_db is not None:
+        loop_db = Path(args.loop_db)
+        if not loop_db.exists():
+            raise ValueError(f"Loop database file does not exist: {args.loop_db}")
+
+    if args.seqres_yaml is not None:
+        seqres_yaml = Path(args.seqres_yaml)
+        # check it exists
+        if not seqres_yaml.exists():
+            raise ValueError(f"SEQRES yaml file does not exist: {args.seqres_yaml}")
+
+    # load the receptor
+
+    receptor_name = receptor.stem
+    logger.info(f"Loaded receptor {receptor_name} from {receptor}")
+
+    xtal = CrystalCompoundData(
+        dataset=Path(args.receptor).stem, str_fn=str(args.receptor)
+    )
+
+    prep_input_schema = PreppedTarget(
+        source=xtal,
+        output_name=str(receptor_name),
+        active_site_chain="A",
+        oe_active_site_residue=None,
+        molecule_filter=MoleculeFilter(
+            components_to_keep=["protein", "ligand"],
+            ligand_chain="A",
+            protein_chains=[],
+        ),
+    )
```

Comment:
i think we will want to make these input args like in the `create_prep_inputs` script:
```
parser.add_argument(
        "--components_to_keep", type=str, nargs="+", default=["protein", "ligand"]
    )
    parser.add_argument("--active_site_chain", type=str, default="A")
    parser.add_argument("--ligand_chain", type=str, default="A")
    parser.add_argument("--protein_chains", type=str, default=[], help="")
    parser.add_argument(
        "--oe_active_site_residue",
        type=str,
        default=None,
        help="OpenEye formatted site residue for active site identification, i.e. 'HIS:41: :A:0: '",
    )
```

Reply:
Yeah apologies this is just me testing.

Reply:
Should have probably waited to push this.